api/controllers/users/profile.ts

```diff
@@ -0,0 +1,135 @@
+declare var sails: any;
+
+module.exports = {
+  friendlyName: 'Fetches user public profile information.',
+  description: 'Basic info as name (firstName, lastName,..), publicBio, \
+    contact data (email, phone), isFreelancer flag. \
+    When service professional, other fields are included as businessName, \
+    websiteUrl (is contact data), profileUrlSlug.',
+  inputs: {
+    id: {
+      example: '141',
+      required: true
+    }
+  },
+  exits: {
+    success: {
+      outputExample: {
+        businessName: 'Loconomics',
+        email: 'iago@loconomics.com',
+        firstName: 'Iago',
+        isClient: true,
+        isOrganization: false,
+        isServiceProfessional: true,
+        lastName: 'Lorenzo Salgueiro',
+        orgDescription: 'Tools for small business owners',
+        orgName: 'Loconomics',
+        orgWebsite: 'https://dev.loconomics.com',
+        phone: '720-555-5555',
+        photoUrl: 'https://dev.loconomics.com/en-US/Profile/Photo/141?v=2018-09-19T18:47:48',
+        publicBio: 'My public bio.',
+        secondLastName: '',
+        serviceProfessionalProfileUrl: 'https://dev.loconomics.com/-iagosrl',
+        serviceProfessionalProfileUrlSlug: 'iagosrl',
+        serviceProfessionalWebsiteUrl: 'https://dev.loconomics.com',
+        timeZone: 'America/Los_Angeles',
+        updatedDate: '2018-09-19T18:47:48.637-07:00',
+        userID: 141
+      }
+    },
+    notFound: {
+      responseType: 'notFound'
+    }
+  },
+  fn: async function(inputs, exits) {
+    const targetUserId = this.req.param('id')
```

Comment:
Out of curiosity, why not use `inputs.id` instead of `req.param`? I'm still learning Sails myself and haven't studied the input code, but I'm not sure I ever see the `id` input ever used anywhere.